src_new/controllers/uas.js

```diff
@@ -0,0 +1,441 @@
+    var phoenixObj = window.Phoenix || {};
+
+    /* istanbul ignore next */
+    refThis.updateSlotsMapFromPhoenixSlots(phoenixObj.getSlots(), [divId], true);
+
+    /* istanbul ignore next */
+    refThis.displayFunctionStatusHandler(getStatusOfSlotForDivId(divId), {}, originalDisplay, [divId]);
+
+    var statusObj = {};
+    statusObj[CONSTANTS.SLOT_STATUS.CREATED] = "";
+
+    refThis.forQualifyingSlotNamesCallAdapters(getSlotNamesByStatus(statusObj), [divId], false);
+
+    /* istanbul ignore next */
+    setTimeout(function() {
+      UTIL.realignVLogInfoPanel(divId);
+      bidManager.executeAnalyticsPixel();
+    }, 2000 + CONFIG.getTimeout());
+
+  });
+}
+/* start-test-block */
+exports.initiateDisplay = initiateDisplay;
+/* end-test-block */
+
+function callJsLoadedIfRequired(win) { // TDD, i/o : done
+    if (UTIL.isObject(win) && UTIL.isObject(win.PWT) && UTIL.isFunction(win.PWT.jsLoaded)) {
+        win.PWT.jsLoaded();
+        return true;
+    }
+    return false;
+}
+/* start-test-block */
+exports.callJsLoadedIfRequired = callJsLoadedIfRequired;
+/* end-test-block */
+
+function createPhoenixNamespace(win){
+	// if Phoenix.isJSLoaded is undefined then it means,
+	// this is first time our JS is loaded on page
+	var Phoenix = win.Phoenix;
+	if( UTIL.isUndefined(Phoenix) || UTIL.isUndefined(Phoenix.isJSLoaded) ){
+		win.Phoenix = new PHOENIX(CONFIG.getTimeout());
+    refThis.initiateDisplay(win);
+	}
+	return win.Phoenix;
+}
+/* start-test-block */
+exports.createPhoenixNamespace = createPhoenixNamespace;
+/* end-test-block */
+
+function initPhoenix(win){
+	if(Phoenix.isJSLoaded == true){
+		return;
+	}
+	Phoenix = refThis.createPhoenixNamespace(win);
+	Phoenix.isJSLoaded = true;
+  UTIL.log("Phoenix.js is loaded successfully.")
+	//setURLs();
+	Phoenix.EQ.executeQ();
+}
+/* start-test-block */
+exports.initPhoenix = initPhoenix;
+/* end-test-block */
+
+function createPubMaticNamespace(win){
+	win.PubMatic = win.PubMatic || {};
+
+  win.PubMatic._uidCB = function(response){
+		PubMatic.pm_uid_bc = response.u;
+		initPhoenix(win);
+	};
+
+	setTimeout(function(){
+		initPhoenix(win);
+	}, 500);
+}
+/* start-test-block */
+exports.createPubMaticNamespace = createPubMaticNamespace;
+/* end-test-block */
+
+//todo: change variable names
+function generateBCUID(win){
+	var c = UTIL.createDocElement(win, "script"),
+		e = win.document.getElementsByTagName("script")[0];
+
+  c.type = "text/javascript";
+	c.async = true;
+	c.src = win.PWT.protocol + "image6.pubmatic.com/AdServer/UCookieSetPug?oid=2&cb=PubMatic._uidCB";
+	e.parentNode.insertBefore(c,e);
+}
+/* start-test-block */
+exports.generateBCUID = generateBCUID;
+/* end-test-block */
+
+exports.init = function(win) { // TDD, i/o : done
+	CONFIG.initConfig();
+    if (UTIL.isObject(win)) {
+        refThis.setWindowReference(win);
+        refThis.wrapperTargetingKeys = refThis.defineWrapperTargetingKeys(CONSTANTS.WRAPPER_TARGETING_KEYS);
```

Comment:
what is purpose of maintaining WrapperTargetingKeys ? in case of DFP we need it to remove key-value pairs added by us in refresh flow